contracts/application/versioning/ImplementationDirectory.sol

```diff
@@ -6,39 +6,43 @@ import 'openzeppelin-solidity/contracts/AddressUtils.sol';
 
 /**
  * @title ImplementationDirectory
- * @dev Concrete version of ImplementationProvider that stores contract implementations in a mapping
+ * @dev Implementation provider that stores contract implementations in a mapping.
  */
 contract ImplementationDirectory is ImplementationProvider, Ownable {
   /**
-   * @dev Emitted when an implementation is added to the directory
-   * @param contractName Name of the contract for which an implementation was added
-   * @param implementation Address of added implementation
+   * @dev Emitted when the implementation of a contract is changed.
+   * @param contractName Name of the contract.
+   * @param implementation Address of the added implementation.
    */
   event ImplementationChanged(string contractName, address implementation);
 
-  // Mapping where the addresses of the implementations are stored
+  /// @dev Mapping where the addresses of the implementations are stored.
   mapping (string => address) internal implementations;
 
   /**
-   * @dev Gets the implementation address for a given contract name
-   * @param contractName Name of the contract whose implementation address is desired
-   * @return Address of the implementation
+   * @dev Returns the implementation address of a contract.
+   * @param contractName Name of the contract.
+   * @return Address of the implementation.
    */
   function getImplementation(string contractName) public view returns (address) {
     return implementations[contractName];
   }
 
   /**
-   * @dev Adds the address of an implementation to the directory
-   * @param contractName Name of the contract whose implementation address is added
-   * @param implementation Address of the implementation
+   * @dev Sets the address of a contract implementation in the directory.
```

Comment:
For consistency with `Freezable`:
"Sets the address of the implementation of a contract."